config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/BundleEntityBuilder.java

```diff
@@ -34,11 +37,137 @@
         this.bundleDocumentBuilder = bundleDocumentBuilder;
     }
 
-    public Element build(Bundle bundle, EntityBuilder.BundleType bundleType, Document document) {
+    public Map<String, Element> build(Bundle bundle, EntityBuilder.BundleType bundleType,
+                                                            Document document, String bundleName,
+                                                            String bundleVersion) {
         List<Entity> entities = new ArrayList<>();
         entityBuilders.forEach(builder -> entities.addAll(builder.build(bundle, bundleType, document)));
 
-        return bundleDocumentBuilder.build(document, entities);
+        List<AnnotatedEntity> annotatedEntities = new ArrayList<>();
+        // Filter the bundle to export only annotated entities
+        // TODO : Enhance this logic to support services and policies
+        final Map<String, Encass> encassEntities = bundle.getEntities(Encass.class);
+        encassEntities.entrySet().parallelStream().forEach(encassEntry -> {
+            Encass encass = encassEntry.getValue();
+            if (encass.getAnnotations() != null) {
+                annotatedEntities.add(createAnnotatedEntity(encass, bundleVersion, encassEntry.getKey()));
+            }
+        });
+
+        if (!annotatedEntities.isEmpty()) {
+            Map<String, Element> annotatedElements = new LinkedHashMap<>();
+            annotatedEntities.stream().forEach(annotatedEntity -> {
+                if (annotatedEntity.isBundleTypeEnabled()) {
+                    List<Entity> entityList = getEntityDependencies(annotatedEntity.getPolicyName(), entities, bundle);
+                    LOGGER.log(Level.FINE, "Entity list : " + entityList);
```

Comment:
Try to reduce the amount of data to be logged here. It is good enough to log included entity details like type, name, id, guid.